### PR TITLE
Add more documents for AR connection_adapters abstract schema_definitions. [ci skip]

### PR DESCRIPTION
**Pull request by JuanitoFatas**
- Add example to column_exists?
- Add example to index_exists?
- Add document for foreign_key
- Add document for foreign_key_exists?

Please review :eyes:, thanks!

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb [hunk 1]
```diff
@@ -535,6 +535,8 @@ def column(column_name, type, options = {})
 
       # Checks to see if a column exists.
       #
+      # t.string(:name) if t.column_exists?(:name, :string)
```
Comment (sgrif) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Wouldn't this be `unless`?

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb [hunk 2]
```diff
@@ -667,10 +673,20 @@ def remove_references(*args)
...
+      # t.foreign_key(:authors)
+      #
+      # See SchemaStatements#add_foreign_key
       def foreign_key(*args) # :nodoc:
         @base.add_foreign_key(name, *args)
       end
 
+      # Checks to see if a foreign key exists.
+      #
+      # t.foreign_key(:authors) if t.foreign_key_exists?(:authors)
```
Comment (sgrif) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Wouldn't this be `unless`?

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb (hunk 1 above)
Comment (JuanitoFatas) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: :+1: Yes.

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb (hunk 2 above)
Comment (JuanitoFatas) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Thanks!

**Comment by JuanitoFatas** [comment]
@sgrif Thanks for review.

**Pull request merged**